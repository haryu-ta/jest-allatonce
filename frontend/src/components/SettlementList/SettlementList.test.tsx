import { render,screen } from "@testing-library/react";
import SettlementList from "./SettlementList";
import { Settlement } from "../../type";


test("SettlementList",() => {
    const settlement : Settlement = {
        from: "Mizuhara",
        to: "Kobayashi",
        amount: 10000
    }
    render(<SettlementList settlements={[settlement]}/>);
    expect(screen.getByText("清算方法")).toBeInTheDocument();
    expect(screen.getAllByRole("listitem").length).toBe(1)
    expect(screen.getByText("Mizuhara → Kobayashi")).toBeInTheDocument();
    expect(screen.getByText("10000円")).toBeInTheDocument();
})
test("SettlementList2",() => {
    const settlement : Settlement[] = [
        {
            from: "Mizuhara",
            to: "Kobayashi",
            amount: 10000
        },
        {
            from: "Mizuhara",
            to: "Suzuki",
            amount: 50000
        }
    ];
    const { container } = render(<SettlementList settlements={settlement}/>);
    expect(container).toMatchSnapshot();
    expect(screen.getByText("清算方法")).toBeInTheDocument();
    expect(screen.getAllByText("清算方法").length).toBe(1);
    expect(screen.getAllByRole("listitem").length).toBe(2)
    expect(screen.getByText("Mizuhara → Kobayashi")).toBeInTheDocument();
    expect(screen.getByText("Mizuhara → Suzuki")).toBeInTheDocument();
    expect(screen.getByText("10000円")).toBeInTheDocument();
    expect(screen.getByText("50000円")).toBeInTheDocument();
})