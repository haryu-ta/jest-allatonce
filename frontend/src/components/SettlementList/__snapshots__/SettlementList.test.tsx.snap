// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettlementList2 1`] = `
<div>
  <div
    class="chakra-stack css-1anxrbq"
  >
    <p
      class="chakra-text css-x39wmu"
    >
      清算方法
    </p>
    <ul
      class="css-1s6xsev"
      role="list"
    >
      <li
        class="css-1v5z18m"
      >
        <div
          class="chakra-stack css-bot922"
        >
          <div
            class="css-0"
          >
            Mizuhara
             → 
            Kobayashi
          </div>
          <div
            class="css-0"
          >
            10000
            円
          </div>
        </div>
      </li>
      <li
        class="css-1v5z18m"
      >
        <div
          class="chakra-stack css-bot922"
        >
          <div
            class="css-0"
          >
            Mizuhara
             → 
            Suzuki
          </div>
          <div
            class="css-0"
          >
            50000
            円
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
